kolibri/core/content/api.py

```diff
@@ -1321,6 +1326,72 @@ def consolidate(self, items, queryset):
         return sorted_items
 
 
+class ContentDownloadRequestViewset(
+    ReadOnlyValuesViewset, CreateModelMixin, DestroyModelMixin
+):
+    serializer_class = serializers.ContentDownloadRequestSeralizer
+
+    pagination_class = OptionalPageNumberPagination
+
+    values = (
+        "id",
+        "requested_at",
+        "reason",
+        "contentnode_id",
+        "metadata",
+        "status",
+        "facility",
+        "source_id",
+    )
+
+    def get_queryset(self):
+        return ContentDownloadRequest.objects.filter(source_id=self.request.user.id)
+
+    def annotate_queryset(self, queryset):
+        return queryset.annotate(
+            has_removal=Exists(
+                ContentRemovalRequest.objects.filter(
+                    source_model=OuterRef("source_model"),
+                    source_id=OuterRef("source_id"),
+                    contentnode_id=OuterRef("contentnode_id"),
+                    requested_at__gte=OuterRef("requested_at"),
+                )
+            )
+        ).filter(has_removal=False)
+
+    def delete(self, request, *args, **kwargs):
+        contentnode_id = request.data.get("contentnode_id")
+
+        if contentnode_id is None:
+            return Response(
+                {"detail": "Contentnode ID is required"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        download_request = ContentDownloadRequest.objects.filter(
+            contentnode_id=contentnode_id,
+            source_id=request.user.id,
+        )
+
+        if download_request.exists():
+            download_request.delete()
```

Comment:
Remove this deletion, and we'll handle it in the removal processing, so that if the removal failed, we can retain visibility on the page for retrying